yt_idefix/data_structures.py

```diff
@@ -361,6 +395,8 @@ def _parse_header_file(self):
         # definitions.h is presumed to be along with data file
         if self._definitions_header is None:
             self._definitions_header = os.path.join(self.fullpath, "definitions.h")
+        elif not os.path.isfile(self._definitions_header):
+            raise FileNotFoundError(f"No such file {self._definitions_header!r}")
 
         if os.path.isfile(self._definitions_header):
```

Comment:
Not quite ! the first check is skipped if `self._definitions_header` was `None` when we entered the function, because we want to hard fail (raise an error) if and _only_ if the user passed an invalid filename, but not if we're just _guessing_ the header filename. Does it make sense ?

Reply:
Yes, I see. 💯  I should be more careful !

Reply:
Or maybe _I_ should comment the code more, depends how you look at it ;-)

Reply:
Hah _a little bit_ more is enough 👍🏼

Reply:
I'll proof-read myself in a couple days when I have completely forgotten how it works again and will add comments where I feel frustrated :-)